### ISSUE TITLE
unintended escaped string for hover

### ISSUE THREAD

**Issue by Hajime-san** [microsoft/TypeScript#57539]
### 🔎 Search Terms

"lsp"

### 🕗 Version & Regression Information

version `"typescript": "5.3.3"`

### 💻 Code

```ts
// 🦁.ts
export const foo = "bar";

// main.ts
import { bar } from './🦁.ts';
```

### 🙁 Actual behavior

The lsp message shows `Module '"./\uD83E\uDD81.ts"' has no exported member 'bar'.ts(2305)`.

### 🙂 Expected behavior

It should be `Module '"./🦁.ts"' has no exported member 'bar'.ts(2305)`.

### Additional information about the issue

<img width="524" alt="スクリーンショット 2024-02-26 10 11 39" src="https://github.com/microsoft/TypeScript/assets/41257923/7c0a935d-5f09-433c-b8af-3b785411b5bd">

**Comment by Andarist** [comment]
This likely falls under the umbrella [here](https://github.com/microsoft/TypeScript/issues/40594) (it's worked on [here](https://github.com/microsoft/TypeScript/pull/49297))

**Comment by fatcerberus** [comment]
@Andarist Wait, is that even related?  That looks like it's about arbitrary import names, e.g.

```ts
import { "😊" as yay } from "foo";
```

...which doesn't work at all today (i.e. it's a feature request), whereas this issue is about unicode characters in module specifiers, which *does* work today, the specifier is just not being rendered property in error messages.

**Comment by Andarist** [comment]
You are totally right. I'm trying to do to much lately and it seems my circuits are overheating :v

**Comment by Andarist** [comment]
My penance: https://github.com/microsoft/TypeScript/pull/57550
